File: ADA/UNIT_1_2/stringmatching.py
def string_matching(text, pattern):
    text_length = len(text)
    pattern_length = len(pattern)

    for i in range(text_length - pattern_length + 1):
        j = 0
        while j < pattern_length:
            if text[i+j] != pattern[j]:
                break
            j += 1

        if j == pattern_length:
            return i  # Pattern found at index i

    return -1  # Pattern not found


text = input("Enter the text: ")
pattern = input("Enter the pattern to search: ")

index = string_matching(text, pattern)

if index != -1:
    print("Pattern found at index", index)
else:
    print("Pattern not found.")

File: ADA/UNIT_1_2/stringmatching.c
#include <stdio.h>
#include <string.h>
int stringMatching(const char *text, const char *pattern)
{
    int textLength = strlen(text);
    int patternLength = strlen(pattern);
    for (int i = 0; i <= textLength - patternLength; i++)
    {
        int j;
        for (j = 0; j < patternLength; j++)
        {
            if (text[i + j] != pattern[j])
                break;
        }
        if (j == patternLength)
            return i;
    }
    return -1;
}
int main()
{
    char text[100];
    char pattern[100];
    printf("Enter the text: ");
    scanf("%[^\n]%*c", text); // Read the whole line including spaces
    printf("Enter the pattern to search: ");
    scanf("%[^\n]%*c", pattern); // Read the whole line including spaces
    int index = stringMatching(text, pattern);
    if (index != -1)
    {
        printf("Pattern found at index %d.\n", index);
    }
    else
    {
        printf("Pattern not found.\n");
    }

    return 0;
}
